### PR TITLE
Usercluster controller manager: Enable leader election

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Enables leader election for the user cluster controller manager to make sure we don't have two versions of the controller manager fighting each others changes during upgrades.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
none
```

/assign @nikhita @zreigz

### CODE REVIEW

**Review by nikhita — APPROVED** [review]
/lgtm
/approve

Fyi @zreigz @p0lyn0mial

File: api/cmd/user-cluster-controller-manager/main.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func main() {
 	// Create Context
 	done := ctx.Done()
 
-	mgr, err := manager.New(cfg, manager.Options{})
+	mgr, err := manager.New(cfg, manager.Options{LeaderElection: true})
```
Comment (p0lyn0mial) [inline api/cmd/user-cluster-controller-manager/main.go:43]: that was easy :)

**Pull request merged**